enterprise_gateway/services/kernels/handlers.py

```diff
@@ -124,4 +124,7 @@ def get(self, kernel_id):
     else:
         # Gen a new type with CORS and token auth
         bases = (TokenAuthorizationMixin, CORSMixin, cls)
+        if cls.__name__ == 'ZMQChannelsHandler':
+            from notebook.base.handlers import APIHandler
+            cls.write_error = APIHandler.write_error
```

Comment:
One of the primary purposes of the gateway's set of mixins is to use `JSONErrorsMixin` for dealing with errors. I'll admit, I'm not up on how all the handlers work (outside of calling the python API for dealing with kernels, etc.), but I think we should either look into injecting JSONErrorsMixin into `bases` or add an override for `ZMQChannelsHandler` like we do for a couple of the kernel handlers.

Also, since `ZMQChannelsHandler` is completely inherited from notebook, why is this change necessary since `APIHandler` is not in its class hierarchy in the first place?  Neither `TokenAuthorizationMixin` or `CORSMixin` implement `write_error()`.

Thanks for your help on this.

Reply:
Mea culpa. I didn't see the existing JSONErrorsMixin solution used above so patched earlier without thinking. Updated as your suggestion. Manually tested again and resolves same issue. Not sure if will cause any regressions in other handlers from notebook_handlers.default_handlers; seems unlikely.

Reply:
Thank you Blair.  Yeah, I think we'll be okay since EG uses very few of the NB handlers.